src/NHibernate/Proxy/ProxyBuilderHelper.cs

```diff
@@ -134,25 +136,48 @@ internal static MethodBuilder GetObjectDataMethodBuilder(TypeBuilder typeBuilder
 			return methodBuilder;
 		}
 
+		private static readonly System.Type[] _equalsParametersTypes = { typeof(object) };
+
 		internal static MethodBuilder GenerateMethodSignature(string name, MethodInfo method, TypeBuilder typeBuilder)
 		{
-			//TODO: Should we use attributes of base method?
-			var methodAttributes = MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.Virtual;
+			var explicitImplementation = method.DeclaringType.IsInterface;
+			if (explicitImplementation &&
+				(typeBuilder.BaseType == typeof(object) ||
+#pragma warning disable 618
+					typeBuilder.BaseType == typeof(ProxyDummy)) &&
+#pragma warning restore 618
+				(method.Name == "Equals" && method.GetParameters().Select(p => p.ParameterType).SequenceEqual(_equalsParametersTypes) ||
+					method.Name == "GetHashCode" && method.GetParameters().Length == 0))
```

Comment:
CodeFactor wants additional redundant parens here while ReSharper does not want them. I am on ReSharper side on this.

Reply:
Please extract 2 methods “IsEquals” and “IsGetHashCode” this way would everyone (CodeFactor, ReSharper and me) would be happy.

Reply:
We could then use them for the other place they are detected and specially handled.